frontend/src/components/admin/ApprovedLanguagesTable.tsx

```diff
@@ -0,0 +1,164 @@
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+
+type ApprovedLanguagesTableProps = {
+  approvedLanguagesTranslation: ApprovedLanguagesMap | undefined;
+  approvedLanguagesReview: ApprovedLanguagesMap | undefined;
+};
+
+interface ApprovedLanguageFieldSortDict {
+  [field: string]: {
+    sortFn: (a: ApprovedLanguage, b: ApprovedLanguage) => number;
+    isAscending: boolean;
+    setIsAscending: Dispatch<SetStateAction<boolean>>;
+  };
+}
+
+type ApprovedLanguage = {
+  language: string;
+  level: number;
+  role: string;
+};
+
+const ApprovedLanguagesTable = ({
+  approvedLanguagesTranslation,
+  approvedLanguagesReview,
+}: ApprovedLanguagesTableProps) => {
+  const [approvedLanguages, setApprovedLanguages] = useState<
+    ApprovedLanguage[]
+  >([]);
+
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+
+  const fieldSortDict: ApprovedLanguageFieldSortDict = {
+    language: {
+      sortFn: generateSortFn<ApprovedLanguage>("language", isAscendingLanguage),
+      isAscending: isAscendingLanguage,
+      setIsAscending: setIsAscendingLanguage,
+    },
+    role: {
+      sortFn: generateSortFn<ApprovedLanguage>("role", isAscendingRole),
+      isAscending: isAscendingRole,
+      setIsAscending: setIsAscendingRole,
+    },
+  };
+
+  const sortApprovedLanguages = (field: string) => {
+    const newApprovedLanguages = [...approvedLanguages];
+    const { sortFn, isAscending, setIsAscending } = fieldSortDict[field];
+    setIsAscending(!isAscending);
+    newApprovedLanguages.sort(sortFn);
+    console.log(newApprovedLanguages);
+    setApprovedLanguages(newApprovedLanguages);
+  };
+
+  useEffect(() => {
+    if (approvedLanguagesTranslation && approvedLanguagesReview) {
+      const translationLanguages = Object.entries(
+        approvedLanguagesTranslation,
+      ).map(([language, level]) => {
+        return { language, level, role: "Translator" };
+      });
+      const reviewLanguages = Object.entries(approvedLanguagesTranslation).map(
+        ([language, level]) => {
+          return { language, level, role: "Reviewer" };
+        },
+      );
+      const combined = translationLanguages.concat(reviewLanguages);
+      sortApprovedLanguages("language");
+      setApprovedLanguages(combined);
+    }
+  }, [approvedLanguagesTranslation, approvedLanguagesReview]);
+
+  const tableBody = approvedLanguages.map(
+    (approvedLanguage: ApprovedLanguage, index: number) => (
+      <Tr
+        borderBottom={
+          index === approvedLanguages.length - 1 ? "1px solid gray" : ""
+        }
+        key={`${approvedLanguage.language}/${approvedLanguage.role}`}
+      >
+        <Td>{convertLanguageTitleCase(approvedLanguage.language)}</Td>
+        <Td>{approvedLanguage.role}</Td>
+        <Td colSpan={4} align="center">
+          <Slider
+            defaultValue={1}
```

Comment:
nit: what happens if you do `defaultValue={approvedLanguage.level}`

Reply:
i'll add an additional field called `sliderValue` to use instead (will help for making the slider functional)